nab_evaluation.py

```diff
@@ -0,0 +1,144 @@
+    @param window_scale_limit (float): The largest amount the windows to be expanded by. For example, for 
+                                       window_scale_limit=2, the new windows will be at most 
+                                       2 * (current window size). 
+    @param goal_sparsity (int): The goal sparsity of the window after increasing the window size. 
+    """
+    if sparsity == 0:  
+        return label_windows
+
+    new_windows = []
+    expand_amount = min(goal_sparsity/sparsity, window_scale_limit) - 1
+    
+    if expand_amount > 0:
+        for i in range(len(label_windows)):
+            start = label_windows[i][0]
+            end = label_windows[i][1]
+            buff_amount = math.ceil(expand_amount/2 * (end - start+1))
+            new_start = max(min_label, start - buff_amount)
+            new_end = min(max_label, end + buff_amount)
+
+            # Merge overlapping windows. Overlapping windows do not happen frequently and 
+            # usually only overlap with adjacent window. 
+            while len(new_windows) > 0 and new_windows[-1][1] >= new_start-1:
+                new_start = min(new_windows[-1][0], new_start)
+                new_end = max(new_windows[-1][1], new_end)
+                del new_windows[-1]
+            new_windows.append((new_start, new_end))
+    else:
+        new_windows = label_windows
+    return new_windows
+
+
+def nab_score(y_true, y_pred):
+    """
+    Computes the NAB score for evaluating the given predictions. 
+    (Ref: https://arxiv.org/ftp/arxiv/papers/1510/1510.03336.pdf)
+    Scoring section (i) handles TP and FN, (ii) handles FP, and TN are 0.
+    (i) Calculate the score for each window. Each window will either have one
+    or more true positives or no predictions (i.e. a false negative). FNs
+    lead to a negative contribution, TPs a positive one.
+    (ii) Go through each false positive and score it. Each FP leads to a negative
+    contribution dependent on how far it is from the previous window.
+    :param y_true: true labels 1/0 for each point
+    :param y_pred: prediction 1/0 for each point by the algorithm
+    :return: nab score
+    """
+    #weights for the standard profile 
+    tp_weight, fp_weight, fn_weight = 1.0, 0.11, 1.0
+    label_windows = label_anomaly_windows(np.array(y_true))
+
+    sparsity = sum(y_true)/float(len(y_true))
+    label_windows = add_buffer_to_label(sparsity, label_windows, 0, len(y_true))
+
+    detection_info = {}
+    for window in label_windows:
+        detection_info[window] = 0
+    tp_score = 0
+    fp_score = 0
+    fn_score = 0
+    for i in range(len(y_pred)):
+        if y_pred[i] == 1:
+            #if index comes before the first window or after last window
+            if i < label_windows[0][0]:
+                fp_score += -1.0*fp_weight
+            elif i > label_windows[-1][1]:
+                position = abs(label_windows[-1][1]-i)/float(label_windows[-1][1]-label_windows[-1][0])
```

Comment:
Based on the latest dicussion, should it be `fp_weight` score?

Reply:
Yeah, you can not give the same score to fp_weight and fn_weight. For standard score, NAB suggests tp_weight = 1.0, fp_weight = 0.11 and fn_weight = 1.0.  This is to compensate for the fact that it is okay to have a couple of false positives if the algorithm predicts some true positives.